### ISSUE TITLE
Setup Status from Granicus (Webhook) - Staging, Prod

### ISSUE THREAD

**Issue by QaysarA** [department-of-veterans-affairs/notification-api#154]
# Value Statement:
**As a** VANotify developer
**I want to** view the status of a notification email
**So that** I get insights into the delivery of the notification

# Acceptance Criteria
1. **GIVEN** an email notification is sent
**WHEN** I access VANotify Dev
**THEN** I view the respective status of the email notification

2. **GIVEN** an email notification is sent
**WHEN** I access VANotify Staging
**THEN** I view the respective status of the email notification

3. **GIVEN** an email notification is sent
**WHEN** I access VANotify Prod
**THEN** I view the respective status of the email notification

# Checklist:
- [ ] ~IP whitelist for Granicus on ALB~ - part of #114
- [x] Datadog graphs
- [x] Review status mapping
- Configure webhook on Granicus side:
  - [x] dev
  - [x] staging
  - [x] production

# Assumptions:
- There is a publicly routable VA url for webhooks for text/email providers
- Granicus contract must be complete for this
- Granicus does not support authentication for its webhooks
-Traffic from dev is being reported to staging - so no Granicus delivery statuses on notifications coming from dev

# Additional Info/Resources:
- https://developer.govdelivery.com/api/tms/resource/webhooks/ See the `POST` request for what we'll have to send to Granicus to configure the webhook on their end

# Out of Scope:
- Status of other notification types like text messages 

# Open Questions:
- How long are the statuses for email notifications stored? 
- Validate mapping the Granicus status to the VANotify status has been documented/completed?
- Will Granicus have three different account/environments (Dev, Staging, Prod) 
- What are the list of statuses we receive from Granicus 
- What happens when if we receive a status that we don't recognize from Granicus. What to we do to follow up from this. Alerts?

**Comment by anushyaprasad** [comment]
blocked by ESSECC request

**Comment by ffafara-tw** [comment]
Answers to open questions:
> How long are the statuses for email notifications stored?
Should not matter as we should be getting webhook calls as soon as status changes.

> Validate mapping the Granicus status to the VANotify status has been documented/completed?
Mapping defined in: app/clients/email/govdelivery_client.py

> Will Granicus have three different account/environments (Dev, Staging, Prod)
only staging and prod. We use Granicus staging in our dev and staging environments

> What are the list of statuses we receive from Granicus
sending, sent, canceled, blacklisted, failed, inconclusive
[https://developer.govdelivery.com/api/tms/resource/webhooks/](https://developer.govdelivery.com/api/tms/resource/webhooks/)

> What happens when if we receive a status that we don't recognize from Granicus. What to we do to follow up from this. Alerts?
Should not happen unless Granicus changes the list of statuses on their side.
It will fail our API validation and return 400 Bad request error to Granicus. Will be visible on our ALB / API 4xx Datadog grapsh.
No alerts.

**Comment by ffafara-tw** [comment]
Screenshot from QA session.
![Screen Shot 2021-03-24 at 3.49.45 PM.png](https://images.zenhubusercontent.com/5da0974a66de7b0001e4d835/39fa2656-e441-4a74-b874-6c1939b950f5)
![Screen Shot 2021-03-24 at 3.49.32 PM.png](https://images.zenhubusercontent.com/5da0974a66de7b0001e4d835/25f0733d-5311-45fa-8a56-9501c42e8b09)
![Screen Shot 2021-03-24 at 3.49.18 PM.png](https://images.zenhubusercontent.com/5da0974a66de7b0001e4d835/5f19f1c4-3c03-455f-b6eb-c0bfe06ac532)
![Screen Shot 2021-03-24 at 3.48.52 PM.png](https://images.zenhubusercontent.com/5da0974a66de7b0001e4d835/08b77479-2920-41d2-969a-04f0602ab278)

**Comment by TWs-DevenS** [comment]
All acceptance criteria have been tested / satisfied (See Screenshots above)
Open Questions have been addressed.
QA Review of Testing completed

**Comment by miabecker** [comment]
Reviewed QA session screen shots and notes and signing off.
